site/src/api/api.ts

```diff
@@ -57,1807 +153,1777 @@ export const withDefaultFeatures = (
+    const response = await this.axios.get(
+      `/api/v2/insights/templates?${searchParams}`,
+    );
+
+    return response.data;
+  };
 
-  try {
-    const res = await axiosInstance.get<TypesGen.JFrogXrayScan>(
-      `/api/v2/integrations/jfrog/xray-scan?${searchParams}`,
+  getHealth = async (force: boolean = false) => {
+    const params = new URLSearchParams({ force: force.toString() });
+    const response = await this.axios.get<TypesGen.HealthcheckReport>(
+      `/api/v2/debug/health?${params}`,
     );
+    return response.data;
+  };
+
+  getHealthSettings = async (): Promise<TypesGen.HealthSettings> => {
+    const res = await this.axios.get<TypesGen.HealthSettings>(
+      `/api/v2/debug/health/settings`,
+    );
+
     return res.data;
-  } catch (error) {
-    if (isAxiosError(error) && error.response?.status === 404) {
-      // react-query library does not allow undefined to be returned as a query result
-      return null;
+  };
+
+  updateHealthSettings = async (data: TypesGen.UpdateHealthSettings) => {
+    const response = await this.axios.put<TypesGen.HealthSettings>(
+      `/api/v2/debug/health/settings`,
+      data,
+    );
+
+    return response.data;
+  };
+
+  putFavoriteWorkspace = async (workspaceID: string) => {
+    await this.axios.put(`/api/v2/workspaces/${workspaceID}/favorite`);
+  };
+
+  deleteFavoriteWorkspace = async (workspaceID: string) => {
+    await this.axios.delete(`/api/v2/workspaces/${workspaceID}/favorite`);
+  };
+
+  getJFrogXRayScan = async (options: GetJFrogXRayScanParams) => {
+    const searchParams = new URLSearchParams({
+      workspace_id: options.workspaceId,
+      agent_id: options.agentId,
+    });
+
+    try {
+      const res = await this.axios.get<TypesGen.JFrogXrayScan>(
+        `/api/v2/integrations/jfrog/xray-scan?${searchParams}`,
+      );
+
+      return res.data;
+    } catch (error) {
+      if (isAxiosError(error) && error.response?.status === 404) {
+        // react-query library does not allow undefined to be returned as a
+        // query result
+        return null;
+      }
+
+      throw error;
+    }
+  };
+}
+
+// This is a hard coded CSRF token/cookie pair for local development. In prod,
+// the GoLang webserver generates a random cookie with a new token for each
+// document request. For local development, we don't use the Go webserver for
+// static files, so this is the 'hack' to make local development work with
+// remote apis. The CSRF cookie for this token is "JXm9hOUdZctWt0ZZGAy9xiS/gxMKYOThdxjjMnMUyn4="
+const csrfToken =
+  "KNKvagCBEHZK7ihe2t7fj6VeJ0UyTDco1yVUJE8N06oNqxLu5Zx1vRxZbgfC0mJJgeGkVjgs08mgPbcWPBkZ1A==";
+
+// Always attach CSRF token to all requests. In puppeteer the document is
+// undefined. In those cases, just do nothing.
+const tokenMetadataElement =
+  typeof document !== "undefined"
+    ? document.head.querySelector('meta[property="csrf-token"]')
+    : null;
+
+interface ClientApi {
```

Comment:
What if instead of having an `api` property, this was `interface ClientApi extends Api`? Then we don't need the `.api` everywhere

Reply:
I considered that – my main concern was organization/discovery, especially with all the functions the file has

At least with the current approach, you know that everything in `api` is an API function, while everything else is something directly relevant to manipulating the client itself

Reply:
This feels like a lot of stuff to have in a completely flat hierarchy
![Screenshot 2024-05-08 at 3 27 44 PM](https://github.com/coder/coder/assets/28937484/3f23cda0-bf77-4f20-92ab-4f95cc3fdd78)

Reply:
yes, but the difference between the "flattened" list and the `.api` list as it is is only 4 functions, and they're 4 functions you hardly ever want to call. it feels odd to me to hide all of stuff you _actually_ want the deepest in the hierarchy, and there are plenty of place's in the go bits of the codebase where we flatten things for exactly this reason.

Reply:
Yeah, I can see that. If we're trying to optimize for the most common use case, then (1) the switch reduces the amount of typing the user needs to do, and (2) it reduces the emphasis placed on methods that are effectively escape hatches

You changed my mind – I'll get this updated tomorrow morning